test/unit/MerkleDrop.spec.js

```diff
@@ -1,7 +1,183 @@
 const { expect } = require("chai");
-const { utils } = require("ethers");
-const { setupFixture, setupInitialState } = require("./utils/prepareState");
+const { utils, BigNumber } = require("ethers");
+const BN = require("bn.js");
 const { ethers } = require("hardhat");
+const {
+  getTranche,
+  createTreeWithAccounts,
+  getAccountBalanceProof,
+} = require("../../tasks/utils/merkle");
+
+const rawAllocations = [
+  "1056.39331099660171709",
+  "870.94684945018301126",
+  "226.48095676324248635",
+  "7.46007341986023537",
+  "9.64320061541387227",
+  "45.45852965164066046",
+  "2.58531507323313161",
+  "105.44076288303354762",
+];
+
+const getParsedAllocations = (addresses, rawAllocations) =>
+  Object.fromEntries(
+    rawAllocations.map((allocation, index) => {
+      return [addresses[index], utils.parseEther(allocation)];
+    })
+  );
+
+const getCumulativeAllocation = (addresses, rawAllocations) => {
+  const parsedAllocations = getParsedAllocations(addresses, rawAllocations);
+
+  return Object.entries(parsedAllocations).reduce(
+    (accumulator, [_, allocation]) => {
+      return accumulator.add(allocation);
+    },
+    BigNumber.from(0)
+  );
+};
+
+const mineBlocks = async (blockAmount) => {
+  for (let i = 0; i < blockAmount; i++) {
+    await network.provider.send("evm_mine");
+  }
+};
+
+const setupFixture = deployments.createFixture(
```

Comment:
If I understand correctly then through the `hardhat-deploy` plugin, you can call `createFixture` which deployed all the contracts you specify and in that way sort of sets up the whole environment. Then you can access the contracts etc through the setupFixture object?

Reply:
In this test file the single specs (`it("blabla", () => {})`) do not build upon each other. In other words, the state of the blockchain is created from scratch for every spec. The advantage is that you can easily add, remove, modify specs without affecting other specs. The disadvantage is that building the state everytime from scratch takes more time. `createFixture` tries to mitigate that issue, by running the deploy scripts once when `setupFixture` is called for the first time, and then for each next time the function is called the state of the blockchain is just reverted back to start from a freshly deployed state without actually having to deploy again. More infos [here](https://github.com/wighawag/hardhat-deploy#creating-fixtures).

I just noticed that this could be further optimized by using `await deployments.fixture(["Migration"]);
` so that only the MerkleDrop contract is deployed.

Reply:
Thanks for the explanation. I'm assuming you always use `createFixture` then? Looking for a best practise for myself, that's why I'm asking the question.

Reply:
Not sure what is best, but I like to have independent/isolated specs and I think then `createFixture` is a good way to achieve that while not making the tests too slow.